packages/signers/src/sign-transaction.ts

```diff
@@ -25,7 +26,10 @@ type CompilableTransactionMessageWithSigners = CompilableTransactionMessage & IT
  */
 export async function partiallySignTransactionMessageWithSigners<
     TTransactionMessage extends CompilableTransactionMessageWithSigners = CompilableTransactionMessageWithSigners,
->(transactionMessage: TTransactionMessage, config: { abortSignal?: AbortSignal } = {}): Promise<NewTransaction> {
+>(
+    transactionMessage: TTransactionMessage,
+    config: { abortSignal?: AbortSignal } = {},
+): Promise<NewTransaction & TransactionWithLifetime> {
```

Comment:
Should we have a helper type like `GetTransactionLifetimeFromTransactionMessage<TTransactionMessage>`?

Then here we can return `Promise<NewTransaction & GetTransactionLifetimeFromTransactionMessage<TTransactionMessage>>` which means TypeScript doesn't loose that information?

Reply:
I tried to do this ([playground](https://www.typescriptlang.org/play?#code/FAFwngDgpgBAkgFQE4EMB2BnFBjEBLAezQFkoMsBzKAdTxAAsARAV1QCMAbKAOSOyg54AZlHwBbWAF4YAb2AwFMQSPFQAXDABMwAL7BQkWIlSYc+IqXIoqtBgCEOBbAGt6KDPQAyw0XgkxpOUUlH1UNAEZdfQNoGABhAjEIPA4UTihkdCxcQhIySil4TNMci3zrGjp6BydXdy9Qv1gAHyKTbPM8qxsqlnYuXjR+ZV8JaPBY4o7cmpc3D28VJoDZeUVOWvn6CKiY2CmzXL60gb4oRdHCoMU0M529PZgAcVED0rQL1QAxJESEpJSJwy7UOZW6UAAPAgAHwBNYKBAwKAADxAUDQABMMPFEslUuk3p1LAUYAB+eHBGAACkRKLRmOxxiyoK6BVs1Ucc3qn2WpNkI1UCUwIFQeDQIA0hJmnLqC0aEh0MA0MgFTSFGBFKDFEpgUqIx3Sg34PIVAEoKcENGgoAA3KBIfRCZhDTowDB4ChoKFI1HorE4gH4rh61kVaFUiTgjT-PFAkPEiqmjQvEAhk1QH5-XGAgkg94JqhQ2HXBRIUSsNAwSMFAB0qv8kkbMHCZP58qg6s12uVGy5HgiOkVyvrHaIGtF4uVtyG6mbg922DHICr5Sos1l2zazPzq8q9hlW3TKxV7YHwHdnoju-XW1NbpQ+AwQjwZDbSwknYnOpDN+57Yei7Ciu4JGrOTIlESu7sgapwziOx4jhomh6BeaBXiBZx3lgj7Pq+J7vqOwpfpKeadDBPBnOmehAA)) but it doesn't seem to want to be that clever

I've added type overloads for the lifetime params instead to mirror what we do with `compileTransactionMessage`, and updated the typetests

Reply:
Oh weird. I thought that's what we were doing with `TClusterUrl` on RPC packages but there must be some subtle change there that makes TS happy. 🤷‍♂️